test/audience.robot

```diff
@@ -27,17 +30,42 @@ Token with correct audience is accepted
     Should Contain   ${out}   404
 
 Token with correct audience in multiple option is accepted
+    [Tags]   audience
     ${uuid}   Generate UUID
     ${se_config}   Get SE config
     ${token}   Get token   scope=-s openid   opts=--aud="${se_config.endpoint} ${uuid}"
     ${url}   SE URL   audience-test-${uuid}
     ${rc}   ${out}   Curl Error   ${url}
     Should Contain   ${out}   404
 
-Token with invalid multiple audiences is rejected
+Token with invalid multiple audiences is rejected (permissive)
+    [Tags]   audience   permissive
+    ${out}   Get token with invalid multiple audience
+    ${ret}   Should Match Regexp   ${out}   40[134]
+
+Token with invalid multiple audiences is rejected (strict)
+    [Tags]   audience   strict
+    ${out}   Get token with invalid multiple audience
+    ${ret}   Should Match Regexp   ${out}   401
+
+
+*** Keywords ***
+
+Get token with random audience
+    ${token}   Get token
+    ${uuid}   Generate UUID
+    ${url}   SE URL   audience-test-${uuid}
+    ${token}   Get token   scope=-s openid   opts=--aud=${uuid}
+    ${rc}   ${out}   Curl Error   ${url}
+    [Return]   ${out}
+
+Get token with invalid multiple audience
+    ${token}   Get token
     ${uuid}   Generate UUID
+    ${url}   SE URL   audience-test-${uuid}
+    ${rc}   ${out}   Curl Put Success   /etc/services  ${url}
+    Should Match Regexp   ${out}   20[01].*
```

Comment:
Is it acceptable practice to do a check in a keyword, rather than in a test? If it isn't, we should probably have to keywords, one for the PUT and one for the GET, both called from the test. Moreover it should be
```suggestion
    Should Match Regexp   ${out}   20[014]
```

Reply:
I agree that those are basically two separate tests in a single one. In order to be more permissive, we can just check that the `GET` method returns `40[134]` removing the bias on the `PUT` request. Moreover, also the original test checked the `GET` request only.